### ISSUE TITLE
Smashing corpses exausting, doesn't scale (enough?) with skill or stats

### ISSUE THREAD

**Issue by Skidborg** [CleverRaven/Cataclysm-DDA#14695]
Here's the problem right now: Pulping corpses drains a ton of stamina, regardless of skill, tools, or strength. Some basic testing with a strength 20 character, 10 bashing skill, 10 melee skill, and a sledgehammer ends up with an empty stamina pool after 10 zombie corpses, and pulping a single zombie hulk consumes more than the entire pool and ~20 in-game minutes.

There doesn't seem to be an advantage over butchery, and it seems like a skilled survivor ought to have a basic understanding of where to cut or crush a zombie to keep it from rising from the dead without dawdling over it for longer than it takes to make a sandwich.

I'd burn zombies instead... but they have to be completely annihilated by roaring flames to prevent reanimation.

**Comment by Coolthulhu** [comment]
Smashing scales badly with bashing damage.
But still, even with optimal weapons (high cutting, low attack time), it could use a buff.

Though it's also about butchery being too good. I nerfed it once or twice (back then it was faster than smashing at full stamina), but was pretty conservative with the nerfing.

**Comment by Skidborg** [comment]
Perhaps it at least needs some interface cues.  It's currently done with the "smash" button, and leaves the corpse "pulped" which one would expect to be entirely dependent on blunt trauma.  The typical assumption is that if you crush the skull and everything inside it, a zombie won't be moving no more.

**Comment by Malkeus** [comment]
Cutting off the head should be a thing, I've got this spiffy battle axe just lying around. Or the good old double tap to the cranium once they are down. Or a special attack with guns at close range which destroys the brain. Does destroying the brain even work on blob zombies? What's the lore on that... I don't smash as a rule, especially late game, because a pulped corpse can still be revived by a necromancer.

This stream of conciousness brought to you by Redbull, in association with NoSleep™

**Comment by hitbutton** [comment]
This is maybe a bit of a tangent, but the biggest issue I have with smashing corpses currently is that, unlike butchering, there's no way to tell the difference between a pulped/unpulped corpse without examining the tile.

**Comment by Malkeus** [comment]
You can use V with a priority on 'corpse' and it will list all the corpses at the top. Yellow ones can revive on their own.

**Comment by hitbutton** [comment]
Thats still a lot more effort than just looking at the screen, as with butchery.

**Comment by Malkeus** [comment]
meh. Back to the topic...why is the bonus to smashing given from cutting weapons? Bash should give the biggest bonus or the result should be a dismembered corpse. If you were literally bashing a corpse to a pulp as seems to be the case, it SHOULD be very tiring and the current setup is fine. But if you're cutting a corpse, I think one would tend to lop off the head and limbs and be done, which would take a lot less effort than smashing the corpse. In which case the current cutting bonus would make sense. 

TLDR:My suggestion: Swap cutting for bash and leave stamina cost alone or Leave cutting bonus as is, reduce stamina cost and make dismembering a thing.

**Comment by kevingranade** [comment]
The idea is the survivor either smashes or dismembers the corpse based on
the weapon used.
If you can come up with a word that describes either smashing or
dismembering, we'll use that.  I'd rather not change the description based
on the weapon used, but it's possible.
As for smashing the head, dda zombies specifically don't care about the
brain more than other parts of the body.

The main thing missing on this discussion is how much effort it *should*
take to do the deed, everyone agrees it takes too much effort, but not how
much.

I'm not at all sure what real life activity would correspond to pulping,
the only thing that comes to mind is base it on butchering but with a
discount for the fact that you don't care about  what happens to the meat.

Something else that comes to mind is making smashing with a blunt weapon
fast but drain more stamina, and make cutting weapons slower but burn less
stamina.

**Comment by Malkeus** [comment]
[This](https://www.youtube.com/watch?v=CEOA884bcu4) is what I picture in my head when smashing a corpse with a blunt object, which is why I consider the stamina cost ok as is. It should be incredibly difficult to perform with a small blunt object like a hammer, small stamina cost per swing, but a hell of a lot of swings to get the job done. A heavy object would do the job faster, taking more stamina per swing, but fewer swings overall.

And something like (Don't Watch this if you're squemish about butchering animals)[this](https://youtu.be/WYancLuBrmE?t=147) but with less care taken to recover usable meat, like you said Kevin. I agree, cutting should take much less stamina, but maybe it would scale based on the weight of the blade and your strength. Using a knife is going to be really slow, unless perhaps you were an MD and knew exactly where to cut, but if you had a very sharp sword you could get a swing on, it should be much faster to hack off limbs. 

Bashing would be tiring, no matter what the end result you're going for. Cutting would be more precise and easier. Either one should get a bonus from MD(knowledge of anatomy), surivival(butchered a lot of corpses), medical(again, knowledge of anatomy). 

If the head is unimportant, what is the minimum damage necessary to make a corpse unrevivable? What is the end goal? Is it complete muscle destruction, or merely severing tendons to render the zombie incapable of moving?

**Comment by Coolthulhu** [comment]
I wanted to do some math regarding the pulp times and I found this:

```
// numbers logic: a str 8 character with a butcher knife (4 bash, 18 cut)
// should have at least a 50% chance of damaging an intact zombie corpse (75 volume).
// a str 8 character with a baseball bat (28 bash, 0 cut) should have around a 25% chance.
```

But that's outdated: corpses got a giant boost to volume quite a while ago. A medium corpse is now 250 volume.

Then there's the part that's still relevant:
```
double pulp_power = sqrt((double)(p->str_cur + p->weapon.type->melee_dam)) *
    std::min(1.0, sqrt((double)(cut_power + 1)));
pulp_power = std::min(pulp_power, (double)p->str_cur);
pulp_power *= 20;
```

`std::min(1.0, something_more_than_1)`?
Looks like cutting has zero effect on pulping right now.

But even after fixing this, that formula would have some issues:
* Sledgehammer would have 7 pulping power at 13 strength
* Crowbar would have >7 pulping power at just 11 strength because it has 1 cutting
* Wooden spear would require only 2 strength to get 7 pulping power

The formula would be very biased against purely bashing weapons, making "pulping" quite impractical.

**Comment by kevingranade** [comment]
I think we're on the same page in general.

The goal of pulping is sheer tissue damage, basically the blob is going to
be able to knit tissue back together to a huge extent, so you have to do
enough damage to make it give up.

It strikes me (pun intended, always) that dismembering should scale better
with skill as well, possibly to the point where bashing is clearly better
at low skill, and dismembering is superior at high skill.

Side issue:
We could add some description to zombie corpses to make it clear that they
are still visibly alive after being downed, and some verbiage when pulping
to make it clear that it has stopped moving entirely.

**Comment by Skidborg** [comment]
The other thing to scale pulping against is combat.  If you're armed well enough to reduce a zombie to unrezzable chunks in one hit, should it take any more than that to pulverize a zombie corpse?

**Comment by i2amroy** [comment]
>As for smashing the head, dda zombies specifically don't care about the
brain more than other parts of the body

Well they utilize the nervous system IIRC, so they do care somewhat since the head is a large concentration of nervous tissue (and is why critical hits are headshots); it's just that they don't care about it *enough* for it to totally destroy them.

That said definitely a +1 to reworking that formula (and maybe to nerfing butchery a bit again). The description idea isn't a bad one either, and cold probably be done fairly easily (just link the ressurects tag into inserting a flag onto the corpse that modifies the description and is purged when the corpse is badly damaged enough to no longer be able to revive).
